src/dotnet/APIView/APIViewWeb/Controllers/AutoReviewController.cs

```diff
@@ -0,0 +1,43 @@
+﻿using APIViewWeb.Filters;
+using APIViewWeb.Respositories;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.FileProviders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace APIViewWeb.Controllers
+{
+    [TypeFilter(typeof(ApiKeyAuthorizeAsyncFilter))]
+    public class AutoReviewController : Controller
+    {
+        private readonly ReviewManager _reviewManager;
+
+        public AutoReviewController(ReviewManager reviewManager)
+        {
+            _reviewManager = reviewManager;
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> UploadAutoReview([FromForm] IFormFile file)
+        {
+            if (file != null)
+            {
+                using (var openReadStream = file.OpenReadStream())
+                {
+                    var review = await _reviewManager.CreateMasterReviewAsync(User, file.FileName, "Auto Review", openReadStream, false);
+                    if(review != null)
+                    {
+                        //Return 200 OK if last revision is approved and 201 if revision is not yet approved.
```

Comment:
How would we use this value?

Reply:
Not sure I understood your question. But this will be checked in caller script in CI to identify the status of review.

Reply:
Regular scheduled CI will not differentiate this status but check that will be added in release stage will check if status is 200 OK or 201 to find if review is approved or now.

Reply:
Would the release check try to upload a new revision?

Reply:
Do we need another method that only compares revisions instead of trying to add one?

Reply:
Seems there are 2 distinct scenarios here.

Reply:
Release step will upload the package. If it finds a diff from last revision it will create a revision and flag CI as review not approved. If it finds same as last revision then it will not create new revision. If last revision is not approved then It will just flag CI as unapproved review. If release stage skips creating a revision when there is a diff from last revision, it will be confusing to reviewers and release approvers why did it flag and it will also miss to show diff it found. To summarize, this will act as a safeguard but in 99% cases release stage will not create a revision.

Reply:
Sounds good